src/battle_main.c

```diff
@@ -571,6 +572,19 @@ static void CB2_InitBattleInternal(void)
     for (i = 0; i < PARTY_SIZE; i++)
         AdjustFriendship(&gPlayerParty[i], FRIENDSHIP_EVENT_LEAGUE_BATTLE);
 
+    // Apply party-wide start-of-battle form changes
+    for (i = 0; i < PARTY_SIZE; i++)
+    {
+        // Player's side
+        targetSpecies = GetFormChangeTargetSpecies(&gPlayerParty[i], FORM_BATTLE_BEGIN, 0);
+        if (targetSpecies != SPECIES_NONE)
+            SetMonData(&gPlayerParty[i], MON_DATA_SPECIES, &targetSpecies);
+        // Opponent's side
+        targetSpecies = GetFormChangeTargetSpecies(&gEnemyParty[i], FORM_BATTLE_BEGIN, 0);
+        if (targetSpecies != SPECIES_NONE)
+            SetMonData(&gEnemyParty[i], MON_DATA_SPECIES, &targetSpecies);
```

Comment:
When setting species, you need to use `CalculateMonStats`, otherwise the stats stay from the previous form:
![image](https://user-images.githubusercontent.com/2904965/187895561-88aaff5a-ddc4-402f-af19-fab7a170b1cb.png) ![image](https://user-images.githubusercontent.com/2904965/187895589-ebbd7299-8699-4ad8-a3aa-dd3aefa62e89.png)
Zacian should've had improved attack and speed but it isn't.